src/go/rdctl/cmd/snapshot.go

```diff
@@ -1,15 +1,142 @@
 package cmd
 
 import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"syscall"
+	"time"
+
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/client"
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/config"
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/paths"
 	"github.com/spf13/cobra"
 )
 
+type cobraFunc func(cmd *cobra.Command, args []string) error
+
 var snapshotCmd = &cobra.Command{
 	Use:    "snapshot",
 	Short:  "Manage Rancher Desktop snapshots",
 	Hidden: true,
+	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+		paths, err := paths.GetPaths()
+		if err != nil {
+			return fmt.Errorf("failed to get paths: %w", err)
+		}
+		// exclude snapshots directory from time machine backups if on macOS
+		if runtime.GOOS == "darwin" {
+			cmd := exec.Command("tmutil", "addexclusion", paths.Snapshots)
+			if err := cmd.Run(); err != nil {
+				return fmt.Errorf("failed to add exclusion to TimeMachine: %w", err)
+			}
+		}
+		return nil
+	},
 }
 
 func init() {
 	rootCmd.AddCommand(snapshotCmd)
 }
+
+func stopBackendCallFuncAndRestartBackend(wrappedFunction cobraFunc) cobraFunc {
+	return func(cmd *cobra.Command, args []string) error {
+		paths, err := paths.GetPaths()
```

Comment:
I get a `Variable 'paths' collides with imported package name` warning -- should use a different name for `paths`, like `appPaths`